src/blog/2024/01/dashboard-2-multi-user.md

```diff
@@ -0,0 +1,127 @@
+
+This new feature will allow you to build Dashboards that provide unique data to each user, build admin-only views, and track user activity, to name but a few. We're really excited to see what the Node-RED Community and our FlowFuse customers can do with such a powerful and flexible framework.
+
+<!--more-->
+
+## Multi User Dashboards
+
+The original Node-RED Dashboard was built with a "single source of truth", no matter how many users interacted with the Dashboard, each user would always see the same data. This is great for prototyping, or hobby projects, but as you scale up your Node-RED usage, you'll want to be able to have unique Dashboard experiences for each user.
+
+### Getting Started
+
+To enable multi-user Dashboards, you'll need to be using FlowFuse, and have an instance with _"FlowFuse User Authentication"_ enabled in the "Security" Settings, this will require any user that wants access to your Editor or Dashboard to be authorized by FlowFuse first.
+
+!["Screenshot of the 'Security' settings available for any Instances running in FlowFuse"](./images/multi-user-dashboard-ffauth.png "Screenshot of the 'Security' settings available for any Instances running in FlowFuse"){data-zoomable}
+<figcaption class="-mt-6 mb-4 text-center"><b>"Screenshot of the 'Security' settings available for any Instances running in FlowFuse"</b></figcaption>
+
+Once enabled, any messages emitted by a Dashboard 2.0 node will contain a new `msg._client.user` object, e.g:
+
+```json
+{
+    "userId": "",   // unique identifier for the user
+    "username": "", // FlowFuse Username
+    "email": "",    // E-Mail Address connected to their FF account
+    "name": "",     // Full Name
+    "image": ""     // User Avatar from FF
+}
+```
+
+Then, when running Node-RED Dashboard 2.0 on FlowFuse, you'll have a new sidebar option in the Node-RED Editor, which allows you to control which node types allow for "client constraints". 
+
+![The new 'FF Auth' options available in Node-RED to allow for client constraints](./images/multi-user-dashboard-ff-settings.png "The new 'FF Auth' options available in Node-RED to allow for client constraints"){data-zoomable}
+<figcaption class="-mt-6 mb-4 text-center"><b>A screenshot of the new 'FF Auth' options available in Node-RED to allow for client constraints on different node types.</b></figcaption>
+
+In the original Node-RED Dashboard, this was _always_ enabled for the `ui-notification` and `ui-control` nodes, whereby you could include `msg.socket` data and it would only then send that message to the specified client. For Dashboard 2.0 we've extended this concept so that as a Node-RED Developer, you can now include `msg._client.user` data in any message sent to a Dashboard 2.0 node. Under the covers, our FlowFuse-exclusive plugin will then automtically filter messages to only send to the relevant user's connection.
```

Comment:
```suggestion
In the original Node-RED Dashboard, this was _always_ enabled for the `ui-notification` and `ui-control` nodes, whereby you could include `msg.socket` data and it would only then send that message to the specified client. For Dashboard 2.0 we've extended this concept so that as a Node-RED Developer, you can now include `msg._client.user` data in any message sent to a Dashboard 2.0 node. Under the covers, our FlowFuse-exclusive plugin will then automatically filter messages to only send to the relevant user's connection.
```